daemon/logger/awslogs/cloudwatchlogs.go

```diff
@@ -80,6 +83,28 @@ func init() {
 	}
 }
 
+func getStreamName(ctx logger.Context) (name string, err error) {
+	if ctx.Config[logStreamKey] != "" {
+		name = ctx.Config[logStreamKey]
+	} else if ctx.Config[logStreamTemplateKey] != "" {
+		var t *template.Template
+		t, err = template.New("t").Parse(ctx.Config[logStreamTemplateKey])
+		if err == nil {
```

Comment:
Got it, the named return was tripping me up.  I'd vote for moving away from the named return and explicitly returning the `err` when it occurs.